### PR TITLE
[MM-55734] Add Request Context to ChannelStore.SaveMember

### PR DESCRIPTION
**Pull request by Alpha-4**
**Fixes:** https://github.com/mattermost/mattermost/issues/26024
**Jira:** [MM-55734](https://mattermost.atlassian.net/browse/MM-55734)

#### Summary
Added **rctx request.CTX**  as mentioned below and did further consequential changes to files.

**Tasks:**
- [x] Change the SaveMember method in server/channels/store/sqlstore/channel_store.go to receive rctx request.CTX as the first argument [1].
- [x] Update the ChannelStore interface in server/channels/store/store.go.
- [x] Make sure all functions calling the changed methods pass along the context.
- [x] Update the generated interfaces with make generated.
- [x] Update the tests.

```release-note
NONE
```

[MM-55734]: https://mattermost.atlassian.net/browse/MM-55734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/channels/store/searchlayer/channel_layer.go [hunk 1]
```diff
@@ -100,11 +100,8 @@ func (c *SearchChannelStore) UpdateMember(rctx request.CTX, cm *model.ChannelMem
 	return member, err
 }
 
-func (c *SearchChannelStore) SaveMember(cm *model.ChannelMember) (*model.ChannelMember, error) {
-	// TODO: Use the actuall request context from the App layer
```
Comment (Alpha-4) [inline server/channels/store/searchlayer/channel_layer.go:104]: removed these comments & assignment as ctx loaded from App.

Comment (hanzei) [reply server/channels/store/searchlayer/channel_layer.go:104]: Yes, thank you :+1:

**Review by hanzei — APPROVED** [review]
Great work, thanks @Alpha-4 :tada:

**Review by fmartingr — APPROVED** [review]
Awesome work, thank you!

**Comment by Alpha-4** [comment]
@hanzei, I see E2E Tests have failed twice on this PR. Do I need to change something? Do let me know if there is some actionable task at my end.

**Comment by Alpha-4** [comment]
@hanzei I have added a new commit after the last merge. Could you please take a look and trigger the tests.

**Comment by hanzei** [comment]
Thank you very much @Alpha-4 :tada:

**Pull request merged**